netsnmp/__init__.py

```diff
@@ -60,14 +60,13 @@ def oid_outside_tree(oid1, oid2):
     Return True if oid2 outside of oid1 tree
     Used by SNMPSession.walk
     """
-    oid1_split = [i for i in oid1.split('.') if i]
-    oid1_idx = len(oid1_split)-1
-
-    oid2_split = [i for i in oid2.split('.') if i]
-
-    if oid2_split[oid1_idx] > oid1_split[oid1_idx]:
+    try:
+        if oid2.startswith(oid1):
+            return False
+        return True
+    except:
         return True
-    return False
+
```

Comment:
I will check this in this weekend, but I think the behavior of walk and getnext should be this:
walk('.1.3.6.1.2.1.1'), as I perceive from snmpwalk commands:
>>> main tree: .1.3.6.1.2.1.1
>>> next:         .1.3.6.1.2.1.1.1.0
>>> next:         .1.3.6.1.2.1.1.2.0

if I execute 'snmpwalk -c public -v2c host .1.3.6.1.2.1.1.1.0',  there is only one result - this oid. There isn't any subtree to walk.